xcube/core/gridmapping/coords.py

```diff
@@ -311,6 +324,27 @@ def grid_mapping_to_coords(
     if xy_dim_names:
         _assert_valid_xy_names(xy_dim_names, name='xy_dim_names')
 
+    if reuse_coords:
+        try:
+            # noinspection PyUnresolvedReferences
+            x, y = grid_mapping.x_coords, grid_mapping.y_coords
+        except AttributeError:
+            x, y = None, None
+        if isinstance(x, xr.DataArray) \
+                and isinstance(y, xr.DataArray) \
+                and x.ndim == 1 \
+                and y.ndim == 1 \
+                and x.size == grid_mapping.width \
+                and y.size == grid_mapping.height:
+            return {
+                name: xr.DataArray(coord.values,
+                                   dims=dim,
+                                   attrs=coord.attrs)
+                for name, dim, coord in zip(xy_var_names,
+                                            xy_dim_names,
+                                            (x, y))
+            }
+
```

Comment:
The line I am referring to in my other comment is the
dtype = np.float64
below. Could this instead be derived from grid_mapping.x_coords' type?

Reply:
I'll fix this in another PR later.